upgrade_tests/helpers/scenarios_utils.py

```diff
@@ -0,0 +1,181 @@
+
+        :param string entities_dict: A dictionary of entities created in
+        satellite
+        """
+    data.update(entities_dict)
+    with open('entities_data', 'wb') as pref:
+        pickle.dump(data, pref)
+
+
+def get_entity_data(scenario_name):
+    """Fetches the dictionary of entities from the disk depending on the
+    Scenario name (class name in which test is defined)
+
+    :param string scenario_name : The name of the class for which the data is
+    to fetched
+
+    :returns dict entity_data : Returns a dictionary of entities
+    """
+    with open('entities_data') as pref:
+        entity_data = pickle.load(pref)
+        entity_data = entity_data[scenario_name]
+    return entity_data
+
+
+def generate_satellite_docker_clients_on_rhevm_upgrade(
+        client_os, clients_count, ak):
+    """Generates satellite clients on docker as containers and subscribes
+        them using provided ak
+
+    :param string client_os: Client OS of which client to be generated
+        e.g: rhel6, rhel7
+    :param string clients_count: No of clients to generate
+    :param string ak : Activation key name, to register clients
+
+    Environment Variables:
+
+    RHEV_SAT_HOST
+        The satellite hostname for which clients to be generated and
+        registered
+    RHEV_CLIENT_AK
+        The AK using which client will be registered to satellite
+    """
+    try:
+        clients_count = int(clients_count)
+        if clients_count == 0:
+            logger.warning(
+                'Clients count to generate on Docker should be atleast 1 !')
+            sys.exit(1)
+        satellite_hostname = get_satellite_host()  # specify satellite hostname
+        result = {}
+        for count in range(clients_count):
+            if bz_bug_is_open('1405085'):
+                time.sleep(5)
+            hostname = '{0}dockerclient{1}'.format(count, client_os)
+            container_id = run(
+                'docker run -d -h {0} -e "SATHOST={1}" '
+                '-e "AK={2}" upgrade:{3}'.format(
+                    hostname, satellite_hostname, ak, client_os))
+            result[hostname] = container_id
+        return result
+    except ValueError:
+        print 'Clients count to generate on Docker should be atleast 1 !'
+
+
+def get_latest_repo_version(cv_name):
+    """Calculates the latest CV version to be published
+
+        :param string cv_name : Name of the CV for which version is to be
+        calculated
+
+        :return int : Calculated version to be created for CV
+        """
+
+    cv_version_data = hammer(
+        'content-view version list --content-view {} '
+        '--organization-id 1'.format(cv_name))
+    latest_cv_ver = sorted([float(data['name'].split(
+        '{} '.format(cv_name))[1]) for data in cv_version_data]).pop()
+    return get_attribute_value(cv_version_data, '{0} {1}'.format(
+        cv_name, latest_cv_ver), 'id')
+
+
+def dockerize(ak_name):
```

Comment:
Ok Let it be, Later we will update this if needed, but I am not happy with duplicating code. We should have updated the existing to match this.